batch/batch/server/server.py

```diff
@@ -295,9 +354,12 @@ def mark_complete(self, pod):
             self._pod_name = None
 
         self._next_task()
-        if self.exit_code == 0 and self._has_next_task():
-            self._create_pod()
-            return
+        if self.exit_code == 0:
```

Comment:
if exit_code == 0 but there are no next tasks, do we want the pvc hanging around?

Reply:
_next_task would have deleted it in this case. I moved the logic here so it is more clear.

Reply:
Ok